vpr/src/place/place_constraints.cpp

```diff
@@ -180,3 +180,36 @@ void load_cluster_constraints() {
         }
     }
 }
+
+void mark_fixed_blocks() {
+    auto& cluster_ctx = g_vpr_ctx.clustering();
+    auto& place_ctx = g_vpr_ctx.mutable_placement();
+    auto& floorplanning_ctx = g_vpr_ctx.floorplanning();
+
+    for (auto blk_id : cluster_ctx.clb_nlist.blocks()) {
+        PartitionRegion pr = floorplanning_ctx.cluster_constraints[blk_id];
+        std::vector<Region> part_region = pr.get_partition_region();
+        if (part_region.size() == 1) {
```

Comment:
possible that you could have two of the same rectangles, could have an area function (or two) for PartitionRegion - one that just checks area, one that checks how many tiles it covers because if it covers more than one tile, you would just break out.